### PR TITLE
remove all clap-related code

### LINKED ISSUE #9: Move clap v3 -> man glue code to clap_generate
As per https://github.com/kbknapp/clap-rs/issues/552#issuecomment-407147885, the clap team would be happy to add the man page generation code to their new `clap_derive` crate. That would mean we could move all clap-specific code there, and only keep the man page generation code in this repo.

cc/ @spacekookie how would you feel about moving code there? I also tagged you in the issue above about joining the clap org.

This furthers the https://github.com/rust-lang-nursery/cli-wg/issues/42 milestone issue.

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Removes all clap related code, in favor of moving it to https://github.com/clap-rs/clap_generate. (Closes #9) Part of a set of 2 PRs to create a demo of Clap + Man working for the 2018 milestone. We should probably hold off merging this until a PR to `clap_generate` is done. Thanks!

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
Okay, I think this is good for review now. Bridged most of the code from @spacekookie with the code I wrote. Created an API that's pretty similar to #2, which should resemble the `clap` API a lot :sparkles:

**Review by killercup — APPROVED** [review]
Awesome! Can't wait for this to be part of my regular clap setup :)

(There are probably quite a few API details to iron out and abstraction we could invent, but if the general use-case is "the user doesn't need to interact with this at all and just gets man pages" we can easily iterate on this)

**Comment by yoshuawuyts** [comment]
`v0.1.0`

File: src/man.rs [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    }
+  }
+  page.section("ENVIRONMENT", &arr)
+}
+
+/// Create a `EXIT STATUS` section.
+///
+/// ## Implementation Note
+/// This currently only returns the status code `0`, and takes no arguments. We
+/// should let it take arguments.
```
[the remark this answers is not in the document]
Comment (killercup) [reply src/man.rs]: Sorry to comment to so late -- yes, "Rust", of course, still trying to teach my phone I'd rather to Rust than rest ;)

Panic handlers like human-panick don't call process::exit so I think they too will exit with 101.

**Comment by intgr** [comment]
Did this feature get removed from clap_generate? I can no longer find documentation for it (https://docs.rs/clap_generate/3.0.0-beta.2/clap_generate/) and there's only an open issue in clap's GitHub https://github.com/clap-rs/clap/issues/552

**Pull request merged**